sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice.json

```diff
@@ -1,3093 +1,3258 @@
+                        "failed",
+                        "cancelled",
+                        "scheduled",
+                        "queued"
+                    ],
+                    "type": "string"
+                },
+                "failureReason": {
+                    "description": "System generated.  Ignored at creation.\r\nIf status == failure, this represents a string containing the reason.",
+                    "type": "string"
+                },
+                "statusMessage": {
+                    "description": "Status message for the job",
+                    "type": "string"
+                },
+                "deviceJobStatistics": {
+                    "$ref": "#/definitions/DeviceJobStatistics",
+                    "description": "Job details"
+                }
+            }
+        },
+        "DeviceJobStatistics": {
+            "description": "The job counts, e.g., number of failed/succeeded devices",
+            "type": "object",
+            "properties": {
+                "deviceCount": {
+                    "format": "int32",
+                    "description": "Number of devices in the job",
+                    "type": "integer"
+                },
+                "failedCount": {
+                    "format": "int32",
+                    "description": "The number of failed jobs",
+                    "type": "integer"
+                },
+                "succeededCount": {
+                    "format": "int32",
+                    "description": "The number of Successed jobs",
+                    "type": "integer"
+                },
+                "runningCount": {
+                    "format": "int32",
+                    "description": "The number of running jobs",
+                    "type": "integer"
+                },
+                "pendingCount": {
+                    "format": "int32",
+                    "description": "The number of pending (scheduled) jobs",
+                    "type": "integer"
+                }
+            }
+        },
+        "QueryResult": {
+            "description": "The query result.",
+            "type": "object",
+            "properties": {
+                "type": {
+                    "description": "The query result type.",
+                    "enum": [
+                        "unknown",
+                        "twin",
+                        "deviceJob",
+                        "jobResponse",
+                        "raw",
+                        "enrollment",
+                        "enrollmentGroup",
+                        "deviceRegistration"
+                    ],
+                    "type": "string"
+                },
+                "items": {
+                    "description": "The query result items, as a collection.",
+                    "type": "array",
+                    "items": {
+                        "type": "object"
+                    }
+                },
+                "con
```

Comment:
Who should be setting all of this? SDK? They should be readonly though.

Reply:
for this specific value, this is the creation time of the message. So user would set it. 
SDK could set it to current time, if the user doesn't supply it. 

The checks on what are required, and what aren't, is something we'll be asking the service folks to confirm and mark in the swagger doc.